### PR TITLE
feat: Spinner, Axios login and routes authorize (#41251)

### PR DESCRIPTION
**Pull request by AryanSinghal**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,7 +9,9 @@
     "@testing-library/jest-dom": "4.2.4",
     "@testing-library/react": "9.4.1",
     "@testing-library/user-event": "7.2.1",
+    "axios": "^0.19.2",
     "commitizen": "^4.0.3",
+    "dotenv": "^8.2.0",
```
Comment (Rahul-Successive) [inline package.json:14]: why are you using dotenv?

File: src/Layout/components/Navbar/Navbar.jsx [hunk 2]
```diff
@@ -1,29 +1,45 @@
...
+              </Typography>
+            </Grid>
+            <Grid item xs={6}>
+              <Button href="/trainee" color="inherit" variant="text">TRAINEE</Button>
+              <Button href="/textfield-demo" color="inherit" variant="text">TEXTFIELD DEMO</Button>
+              <Button href="/input-demo" color="inherit" variant="text">INPUT DEMO</Button>
+              <Button href="/children-demo" color="inherit" variant="text">CHILDREN DEMO</Button>
+              <Button onClick={this.handleLogout} color="inherit" variant="text">LOGOUT</Button>
+              {
+                (!localStorage.getItem('token'))
```
Comment (Rahul-Successive) [inline src/Layout/components/Navbar/Navbar.jsx]: What is this?

File: src/pages/Login/Login.jsx [hunk 3]
```diff
@@ -72,70 +77,119 @@ class Login extends React.Component {
...
-            <div align="center">
-              <Avatar className={classes.avatar}>
-                <LockIcon />
-              </Avatar>
-            </div>
-            <br />
-            <Typography align="center" gutterBottom variant="h5">
-              Login
+      <>
+        {
```
Comment (Rahul-Successive) [inline src/pages/Login/Login.jsx]: Again why it is there?

File: src/routes/PrivateRoute.jsx [hunk 4]
```diff
@@ -1,17 +1,24 @@
...
-  <Route
-    {...rest}
-    render={(matchProps) => (
-      <PrivateLayout>
-        <Component {...matchProps} />
-      </PrivateLayout>
-    )}
-  />
+  <>
+    {
```
Comment (Rahul-Successive) [inline src/routes/PrivateRoute.jsx]: Move this code outside of JSX Scope

**Review by Rahul-Successive — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (AryanSinghal) [reply package.json:14]: Because we are said to do so

File: src/Layout/components/Navbar/Navbar.jsx (hunk 2 above)
Comment (AryanSinghal) [reply src/Layout/components/Navbar/Navbar.jsx]: I have made the changes

File: src/pages/Login/Login.jsx (hunk 3 above)
Comment (AryanSinghal) [reply src/pages/Login/Login.jsx]: I have made the changes

File: src/routes/PrivateRoute.jsx (hunk 4 above)
Comment (AryanSinghal) [reply src/routes/PrivateRoute.jsx]: ok sir

**Review by Rahul-Successive — APPROVED** [review]

**Pull request merged**